packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -584,12 +585,19 @@ const DataGrid = injectIntl((props) => {
       // Not triggered when swapping focus between children
       if (handleFocus.current) {
         setFocusedRowCol(focusedRow, focusedCol, true);
+        setGridHasFocus(true);
```

Comment:
I would think that we could determine this information without needing to alter the focus event handler logic.